util/Setup/Program.cs

```diff
@@ -218,26 +226,61 @@ private static bool ValidateInstallation()
         }
         else
         {
-            installationKey = Helpers.ReadInput("Enter your installation key");
+            var prompt = "Enter your installation key";
+            installationKey = Helpers.ReadInput(prompt);
+            while (installationKey == string.Empty)
+            {
+                Helpers.WriteError("Invalid input for installation key. Please try again.");
+                installationKey = Helpers.ReadInput(prompt);
+            }
+        }
+
+        if (_context.Parameters.ContainsKey("cloud-region"))
```

Comment:
I generally prefer `_context.Parameters.TryGetValue("cloud-region", out var cloudRegion)` because you access the dictionary only once and you save yourself from having the keys differ in both their uses, but I see that isn't the pattern here so I will leave it up to you.

Reply:
I think I'd like to continue following the pattern here instead of refactoring all of it this time.